fuse_constraints/include/fuse_constraints/variable_constraints.h

```diff
@@ -98,7 +98,7 @@ class VariableConstraints
    * Accessing a variable id that is not part of this container results in undefined behavior
    */
   template <typename OutputIterator>
-  void getConstraints(const unsigned int variable_id, OutputIterator result) const;
+  void getConstraints(const unsigned int variable_id, OutputIterator&& result) const;
```

Comment:
Maybe it makes more sense to have two implementations:

- One with `OutputIterator` that takes an rvalue by copy or with `OutputIterator&&` using move semantics. This will support `std::back_inserter`
-  Another one with `OutputIterator&` that takes an lvalue by reference. This will support https://github.com/locusrobotics/fuse/pull/133

In the case of `OutputIterator&` we also need `result = std::copy(...);`, but not in the other case.

Another option would be to change the `getConstraints` API to follow the one of `std::copy`, so it returns `result`, i.e. `return std::copy(...);`, in which case the `OutputIterator` arguments would always be passed by copy. :thinking:

Reply:
Based on the signature of `std::copy`, I think the correct thing to do is return the "one passed the last" iterator from the function.
```
template< class InputIt, class OutputIt >
OutputIt copy( InputIt first, InputIt last, OutputIt d_first );
```
And then fix up the usage in the marginalize code.

Thanks for catching this.

Reply:
Done (in my last commit)